### ISSUE TITLE
[Bug]: The http protocol was transformed to https when specifying headless: 'new' or false

### ISSUE THREAD

**Issue by zzp-linking** [puppeteer/puppeteer#10791]
### Minimal, reproducible example

```TypeScript
Internal network environment, I can't copy my code here. I can show some photo with mosaic。
```

### Error string

no error

### Bug behavior

- [ ] Flaky
- [ ] PDF

### Background

Internal network environment, I can't copy my code here. I can show some photo with mosaic。

when i use puppeteer to screenshot my webpage, i always receive a empty photo.
Then when i set headless to false, i found that my page protocol is http, but the chrome open my page with https protocol.
Only oen website was transformed, otheres is ok. when i set headless to 'old', all website performance ok!
details:
1. I use postman send a request, and my url is startsWith http; 
![1.jpg](https://i.postimg.cc/cH8NM8yf/1.jpg)
2. my program received request and get the url param is startsWith http; 
![2.jpg](https://i.postimg.cc/7h8rBYwd/2.jpg)
3. the puppeteer open my page with https protocol;
![3.jpg](https://i.postimg.cc/dVdtvXS8/3.jpg)
4. i use puppeteer open another empty page, address is localhost:8082/xxxxx, content is loading, then i execute code: window.location.href = 'http://xxx.xxx.com/xxx.html#/system/reset-password';
![4.jpg](https://i.postimg.cc/BQ54VD8g/4.jpg)
5. the page go to https://xxx.xxx.com/xxx.html#/system/reset-password;
![5.jpg](https://i.postimg.cc/Dypp19XL/5.jpg)
6. from the Fiddler, i found there is no any redirect when i execute the step4 code, it open a https page directly;
![6.jpg](https://i.postimg.cc/BZrny4mq/6.jpg)
7. when i pass the http address on the browser address-bar directly (with out puppeteer)，it open my webpage correctly；
![7.jpg](https://i.postimg.cc/0jFP1dYW/7.jpg)

### Expectation

don't transform my url protocol

### Reality

pupeteer transformed the url protocol from http to https

### Puppeteer version

21.1.0

### Node version

18.7.0

### Package manager

npm

### Package manager version

8.15.0

### Operating system

Windows

**Comment by OrKoN** [comment]
Puppeteer does not convert the URLs to HTTPS. It's the browser's behaviour as you confirmed by using the old headless mode (see https://developer.chrome.com/articles/new-headless/).

**Comment by zzp-linking** [comment]
OK，thanks

**Comment by bmax** [comment]
@OrKoN @zzp-linking Is there a way to disable this?

**Comment by OrKoN** [comment]
I believe it is `--disable-features=HttpsUpgrades` but eventually the behavior will probably be the default in Chrome without an option to turn it off.

**Comment by bmax** [comment]
thanks friend.

**Comment by asterite3** [comment]
> I believe it is `--disable-features=HttpsUpgrades` but eventually the behavior will probably be the default in Chrome without an option to turn it off.

That's very helpful, thanks!

Why do you think the option to turn off the behavior will be removed in future? The option is useful for some scenarios (including automated testing, not for normal browser usage).

**Comment by asterite3** [comment]
In case someone is struggling with this now: solely using `--disable-features=HttpsUpgrades` did not work for me, but I managed to disable auto-chaging URL scheme to https by disabling both `HttpsUpgrades` and `HttpsFirst*` feature flags (at least `HttpsFirstBalancedModeAutoEnable`). This flag is [mentioned by Puppeteer docs](https://pptr.dev/troubleshooting#neterr_blocked_by_client-when-navigating-to-an-http-url-in-chrome). Tested on Chrome for  Testing 135.0.7049.84.
